balance_strategy_test.go

```diff
@@ -100,3 +105,2198 @@ func TestBalanceStrategyRoundRobin(t *testing.T) {
+		t.Error("stickyBalanceStrategy.Plan() consumer1 didn't maintain partitions across reassignment")
+	}
+	if len(intersection(plan1["consumer2"]["topic1"], plan2["consumer2"]["topic1"])) != 2 {
+		t.Error("stickyBalanceStrategy.Plan() consumer1 didn't maintain partitions across reassignment")
+	}
+
+	// PLAN 3
+	delete(members, "consumer1")
+	members["consumer2"] = ConsumerGroupMemberMetadata{
+		Topics:   []string{"topic1"},
+		UserData: encodeSubscriberPlanWithGeneration(t, plan2["consumer2"], 2),
+	}
+	members["consumer3"] = ConsumerGroupMemberMetadata{
+		Topics:   []string{"topic1"},
+		UserData: encodeSubscriberPlanWithGeneration(t, plan1["consumer3"], 1),
+	}
+
+	plan3, err := s.Plan(members, topics)
+	verifyPlanIsBalancedAndSticky(t, s, members, plan3, err)
+	verifyFullyBalanced(t, plan3)
+}
+
+func Test_stickyBalanceStrategy_Plan_AssignmentWithMultipleGenerations2(t *testing.T) {
+	s := &stickyBalanceStrategy{}
+
+	topics := map[string][]int32{"topic1": []int32{0, 1, 2, 3, 4, 5}}
+	members := map[string]ConsumerGroupMemberMetadata{
+		"consumer1": ConsumerGroupMemberMetadata{Topics: []string{"topic1"}},
+		"consumer2": ConsumerGroupMemberMetadata{Topics: []string{"topic1"}},
+		"consumer3": ConsumerGroupMemberMetadata{Topics: []string{"topic1"}},
+	}
+	plan1, err := s.Plan(members, topics)
+	verifyPlanIsBalancedAndSticky(t, s, members, plan1, err)
+	verifyFullyBalanced(t, plan1)
+
+	// PLAN 2
+	delete(members, "consumer1")
+	members["consumer2"] = ConsumerGroupMemberMetadata{
+		Topics:   []string{"topic1"},
+		UserData: encodeSubscriberPlanWithGeneration(t, plan1["consumer2"], 1),
+	}
+	delete(members, "consumer3")
+
+	plan2, err := s.Plan(members, topics)
+	verifyPlanIsBalancedAndSticky(t, s, members, plan2, err)
+	verifyFullyBalanced(t, plan2)
+	if len(intersection(plan1["consumer2"]["topic1"], plan2["consumer2"]["topic1"])) != 2 {
+		t.Error("stickyBalanceStrategy.Plan() consumer1 didn't maintain partitions across reassignment")
+	}
+
+	// PLAN 3
+	members["consumer1"] = ConsumerGroupMemberMetadata{
+		Topics:   []string{"topic1"},
+		UserData: encodeSubscriberPlanWithGeneration(t, plan1["consumer1"], 1),
+	}
+	members["consumer2"] = ConsumerGroupMemberMetadata{
+		Topics:   []string{"topic1"},
+		UserData: encodeSubscriberPlanWithGeneration(t, plan2["consumer2"], 2),
+	}
+	members["consumer3"] = ConsumerGroupMemberMetadata{
+		Topics:   []string{"topic1"},
+		UserData: encodeSubscriberPlanWithGeneration(t, plan1["consumer3"], 1),
+	}
+	plan3, err := s.Plan(members, topics)
+	verifyPlanIsBalancedAndSticky(t, s, members, plan3, err)
+	verifyFullyBalanced(t, plan3)
+}
+func Test_stickyBalanceStrategy_Plan_AssignmentWithConflictingPreviousGenerations(t *testing.T) {
+	s := &stickyBalanceStrategy{}
+
+	topics := map[string][]int32{"topic1": []int32{0, 1, 2, 3, 4, 5}}
+	members := make(map[string]ConsumerGroupMemberMetadata, 3)
+	members["consumer1"] = ConsumerGroupMemberMetadata{
+		Topics:   []string{"topic1"},
+		UserData: 
```

Comment:
This can take `testing.TB` and then the duplicate function just below can be eliminated.